IOS/SwiftSeedProject/SwiftSeedProject/Application/Service/BaseService.swift

```diff
@@ -7,25 +7,34 @@
 //
 
 import Foundation
+import SwiftyJSON
 
 class BaseService<T: PersistenceObject>  {
     
-    private let persistence: Persistence
-    private let restClient: AnyObject? = nil
-    
-    //TODO: we should inclued the API access also
-    init(persistence: Persistence) {
+    private let persistence: Persistence!
+    internal let api: Api!
+
+    init(persistence: Persistence, api: Api) {
         self.persistence = persistence
+        self.api = api
     }
     
     public func getAll() -> [T] {
         return persistence.getAll(entityName: T.EntityName)
     }
     
-    public func getEntityBy(identifier: Int) -> T {
+    public func getEntityBy(identifier: String) -> T? {
         return persistence.getBy(entityIdentifier: identifier, entityName: T.EntityName)
     }
     
+    public func createEntity(json: JSON) {
+        persistence.add(json: json, entityName: T.EntityName)
+    }
+    
+    public func save() {
+        persistence.save()
```

Comment:
> I think that the service should not has a save method. What do you think?

Exposed as a public method? definitely not. Good catch!

> Another option would be to make the persistence object protected and call save from the persistence directly

Yes, I think that's OK. It might look like an inconsistency, but once this services grow and handle all the domain-related stuff, it will make much more sense. Now it looks like a simple persistence proxy